api/proto/teleport/legacy/types/types.proto

```diff
@@ -7006,29 +7008,9 @@ message JamfSpecV1 {
   // Example: "https://yourtenant.jamfcloud.com/api".
   // Required.
   string api_endpoint = 4 [(gogoproto.jsontag) = "api_endpoint,omitempty"];
-  // Jamf API username.
-  // Username and password are used to acquire short-lived Jamf Pro API tokens.
-  // See https://developer.jamf.com/jamf-pro/docs/jamf-pro-api-overview.
-  // Prefer using client_id and client_secret.
-  // Either username+password or client_id+client_secret are required.
-  string username = 5 [(gogoproto.jsontag) = "username,omitempty"];
```

Comment:
Let's do this in 3 steps:

1. First we add the new JamfCredentials type,
2. then we update e/,
3. then we drop the older fields.

Doing it in 2 steps will break the build.